query-algebrizer/src/clauses/not.rs

```diff
@@ -0,0 +1,450 @@
+        let vx = Variable::from_valid_name("?x");
+
+        let knows = QueryValue::Entid(66);
+        let parent = QueryValue::Entid(67);
+
+        let john = QueryValue::TypedValue(TypedValue::typed_string("John"));
+        let ambar = QueryValue::TypedValue(TypedValue::typed_string("Ámbar"));
+        let daphne = QueryValue::TypedValue(TypedValue::typed_string("Daphne"));
+
+        let d0e = QualifiedAlias::new(d0.clone(), DatomsColumn::Entity);
+        let d0a = QualifiedAlias::new(d0.clone(), DatomsColumn::Attribute);
+        let d0v = QualifiedAlias::new(d0.clone(), DatomsColumn::Value);
+        let d1e = QualifiedAlias::new(d1.clone(), DatomsColumn::Entity);
+        let d1a = QualifiedAlias::new(d1.clone(), DatomsColumn::Attribute);
+        let d1v = QualifiedAlias::new(d1.clone(), DatomsColumn::Value);
+
+        let mut subquery = ConjoiningClauses::default();
+        subquery.from.append(&mut vec![SourceAlias(DatomsTable::Datoms, d0), SourceAlias(DatomsTable::Datoms, d1)]);
+        subquery.column_bindings.insert(vx.clone(), vec![d0e.clone(), d1e.clone()]);
+        subquery.wheres = ColumnIntersection(vec![ColumnConstraintOrAlternation::Alternation(ColumnAlternation(vec![
+                                                    ColumnIntersection(vec![
+                                                        ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d0a.clone(), knows.clone())),
+                                                        ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d0v.clone(), john))]),
+                                                    ColumnIntersection(vec![
+                                                        ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d0a.clone(), knows)),
+                                                        ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d0v.clone(), ambar))]),
+                                                    ])),
+                                                    ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d1a.clone(), parent.clone())),
+                                                    ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d1v.clone(), daphne.clone())),
+                                                    ColumnConstraintOrAlternation::Constraint(ColumnConstraint::Equals(d0e.clone(), QueryValue::Column(d1e.clone())))]);
+
+        subquery.known_types.insert(vx.clone(), ValueTypeSet::of_one(ValueType::Ref));
+
+        assert!(!cc.is_known_empty());
+        assert_eq!(cc.wheres, ColumnIntersection(vec![
+                ColumnConstraintOrAlternation::Constraint(ColumnConstraint::NotExists(ComputedTable::Subquery(subquery))),
+            ]));
+    }
+
+    // not-join.
+    #[test]
+    fn test_not_with_in() {
+        let schema = prepopulated_schema();
+        let query = r#"
+            [:find ?x
```

Comment:
This query should fail! The projection list is not bound inside the query. `not` doesn't bind, so `?x` is unbound.

It should work if you change this to `:find ?y`. Please duplicate this test to do that, and make sure it works, and then make this one fail!